jerry-core/api/jerry.c

```diff
@@ -5281,6 +5281,113 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
+#if ENABLED (JERRY_BUILTIN_CONTAINER)
+  if ((arguments_number >= 1 && ecma_is_value_error_reference (arguments[0]))
+      || (arguments_number >= 2 && ecma_is_value_error_reference (arguments[1])))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  if (!ecma_is_value_object (container))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Container is not an object.")));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (container);
+
+  if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_CLASS)
+  {
+    uint16_t type = ((ecma_extended_object_t *) obj_p)->u.class_prop.class_id;
+    ecma_extended_object_t *container_object_p = ecma_op_container_get_object (container, type);
+    jerry_value_t result;
+
+    if (container_object_p == NULL)
+    {
+      return ecma_create_error_reference_from_context ();
+    }
+
+    switch (operation)
+    {
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET)
+      case JERRY_CONTAINER_OP_ADD:
+      {
+        result = jerry_return (ecma_op_container_set (container_object_p, arguments[0], arguments[0], type));
+        break;
+      }
+#endif /* ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET) */
+#if ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP)
+      case JERRY_CONTAINER_OP_GET:
+      {
+        result = jerry_return (ecma_op_container_get (container_object_p, arguments[0], type));
+        break;
+      }
+      case JERRY_CONTAINER_OP_SET:
+      {
+        result = ecma_op_container_set (container_object_p, arguments[0], arguments[1], type);
+
+        if (!jerry_value_is_error (result))
+        {
+          jerry_release_value (result);
+          result = jerry_create_boolean (true);
+        }
+        break;
+      }
+#endif /* ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP) */
+      case JERRY_CONTAINER_OP_HAS:
+      {
+        result = ecma_op_container_has (container_object_p, arguments[0], type);
+        break;
+      }
+      case JERRY_CONTAINER_OP_DELETE:
+      {
+        if (type == LIT_MAGIC_STRING_WEAKMAP_UL || type == LIT_MAGIC_STRING_WEAKSET_UL)
+        {
+          result = ecma_op_container_delete_weak (container_object_p, arguments[0], type);
+          break;
+        }
+        result = ecma_op_container_delete (container_object_p, arguments[0], type);
+        break;
+      }
+      case JERRY_CONTAINER_OP_SIZE:
+      {
+        result = ecma_op_container_size (container_object_p);
+        break;
+      }
+      case JERRY_CONTAINER_OP_CLEAR:
+      {
+        result = ecma_op_container_clear (container_object_p);
+        break;
+      }
+      default:
+      {
+        result = jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Unsupported container operation")));
+      }
+    }
+    return jerry_return (result);
+  }
+  return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Containers are disabled.")));
```

Comment:
I think the error messages should be swapped
Containers are disabled <-> Container is not an object.

Also, we could do an early return (when `if (ecma_get_object_type (obj_p) != ECMA_OBJECT_TYPE_CLASS)`)

Reply:
This request is still valid. Please recheck the error messages.